main.go

```diff
@@ -18,11 +18,24 @@ package main
 
 import (
 	"fmt"
+	"io"
 	"log"
 	"os"
 )
 
+var (
+	fs     fileSystem
+	runner cmdRunner
+	rc     clock
+	out    io.Writer
```

Comment:
I added these globals which are set in `func main()` and used throughout because they allow us to mock a lot of things in our functions without having to pass them in. Since tests never run `func main()` they will be set to null and will require being set in a test. Depending on what you are testing you can add the concrete class like we do in main or you can add a mock one that we have. Another option would be to create a brand new mock if an existing one doesn't do what you need. Since Go works off duck typing, it's very easy to create a new mock.